src/Thesis/Ast.hs

```diff
@@ -33,3 +37,34 @@ data Select = Select {
   selectFrom :: Identifier,
   selectWhere :: Maybe Expr
 }
+
+selectToQuery :: Select -> (Query, [Parameter])
+selectToQuery (Select sAgg sFrom sWhere) = toQuery $ emit "SELECT " <> emitAggregation sAgg <> emitFrom sFrom <> emitWhere sWhere
+
+emitExpr :: Expr -> SqlPart
+emitExpr (Literal (Value v)) = emit $ pack $ show v
+emitExpr (Column identifier) = emitIdentifier identifier
+emitExpr (PrefixOp identifier expr) = emit identifier <> emitExpr expr
+emitExpr (PostfixOp identifier expr) = emitExpr expr <> emit identifier
+emitExpr (BinaryOp expr1 identifier expr2) = emitExpr expr1 <> emit identifier <> emitExpr expr2
+emitExpr (FunctionCall identifier exprs) = emit identifier <> emit "(" <> mconcat (emitExpr `map` exprs) <> emit ")"
+
+emitAggregation :: Aggregation -> SqlPart
+emitAggregation (Average expr) = emit "AVG(" <> emitExpr expr <> emit ") "
```

Comment:
We should probably adopt a convention on where trailing whitespace is added. I think that the current implementation is inconsistent - a trailining space is added only in `emitAggregation` but no other function.